Atomic/client.py

```diff
@@ -110,3 +112,11 @@ def iter_subs(tree, key=None):
         for i in range(len(tree)):
             tree[i] = iter_subs(tree[i])
     return tree
+
+
+def ids_same(id1, id2):
+    if id1 == id2:
+        return True
+    if no_shaw(id1) == no_shaw(id2):
```

Comment:
Why is this discarding the algorithm identifier? At a first glance I can’t see any reason for this, and it could allow attacks using a weaker algorithm against a stronger one.